docs/source/configuring_deploying.rst

```diff
@@ -0,0 +1,298 @@
+remove nodes to change membership (see `Changing Network Membership`_ for more
+details).
+
+To setup a fully-peered network, all validators must be started with the
+following two options:
+
+* ``--peering static`` - Informs the validator to connect with all of the
+  specified peers.
+* ``--peers {peers_list}`` - Provides the list of peers for the validator to
+  connect to, where `{peers_list}` is the list of public endpoints for all other
+  previously started validators.
+
+For example, if there are two validators already running with public endpoints
+``203.0.113.1:8800`` and ``203.0.113.2:8800``, a new validator would be started
+with the options ``--peering static --peers
+tcp://203.0.113.1:8800,tcp://203.0.113.2:8800``. Since peering is
+bi-directional, when the new validator connects with the existing validators,
+the existing validators will be fully peered as well.
+
+
+On-Chain Settings
+=================
+
+When starting a fresh network with the Sawtooth Raft consensus engine, the
+required settings below (and the desired optional settings) should be set in the
+genesis block.
+
+
+Required Settings
+-----------------
+
+* ``sawtooth.consensus.algorithm`` - Tells the network which consensus engine to
+  use. Must be set to ``raft``.
+* ``sawtooth.consensus.raft.peers`` - A JSON list of public keys of all
+  validators in the network, formatted as hex strings.
+
+The ``sawtooth.consensus.raft.peers`` setting is extremely important, since the
+Raft engine uses it as its source-of-truth for which nodes to communicate with.
+When a Raft engine starts initially, it checks the setting to determine which
+nodes are in the network. This setting is also used to add or remove nodes from
+the Raft network as covered in the `Changing Network Membership`_ section. As an
+example, the value of ``sawtooth.consensus.raft.peers`` would be formatted like
+this:
+
+.. code-block:: json
+
+  [
+    "dc26a7099e81bb02869cc8ae57da030fbe4cf276b38ab37d2cc815fec63a14ab",
+    "df8e8388ced559bd35c2b05199ca9f8fbebb420979715003355dcb7363016c1d"
+  ]
+
+
+Optional Settings
+-----------------
+
+* ``sawtooth.consensus.raft.heartbeat_tick`` - Determines the interval at which
+  the leader sends heartbeat messages to all other nodes. Default is ``2``.
+* ``sawtooth.consensus.raft.election_tick`` - Determines the timeout after which
+  nodes can start new elections if no messages have been received from the
+  leader. Default is ``20``.
+* ``sawtooth.consensus.raft.period`` - Determines the length of time (in
+  milliseconds) that the consensus engine will wait between initializing and
+  finalizing a block. Default is ``3000``.
+
+These optional settings should only be set `before` any Raft engines have
+started up; if they are changed on a running network and a new node is started,
+these values will be different on the new engine than the rest of the network,
+which could result in unpredictable behavior.
```

Comment:
Won't the new node pick up the setting change as it plays forward state?

Reply:
Currently, we don't support changing the heartbeat tick, election tick, or period on a running network. This would require monitoring these on-chain settings like we do with `sawtooth.consensus.raft.peers`.

Reply:
Added improvement to JIRA for supporting on-the-fly configuration changes: https://jira.hyperledger.org/browse/STL-1457